app/controllers/packages_controller.rb

```diff
@@ -0,0 +1,73 @@
+require 'httparty'
+
+class PackagesController < ApplicationController
+ # include Timeout
+
+  def rates
+    origin = get_origin
+    package = get_package
+    destination = get_destination
+
+     ups = ActiveShipping::UPS.new(login: ENV["UPS_LOGIN"], password: ENV["UPS_PASSWORD"], key: ENV["UPS_KEY"])
+     usps = ActiveShipping::USPS.new(login: ENV["USPS_LOGIN"])
+
+    ups_response = ups.find_rates(origin, destination, package)
+    ups_rates = ups_response.rates.sort_by(&:price).collect {|rate| [rate.service_name, rate.price, rate.delivery_date]}
+
+    usps_response = usps.find_rates(origin, destination, package)
+    usps_rates = usps_response.rates.sort_by(&:price).collect {|rate| [rate.service_name, rate.price, rate.delivery_date]}
+
+    all_rates = { ups: ups_rates, usps: usps_rates }
+
+
+    # begin
+    #   Timeout::timeout(10) do
+        if ups_rates && usps_rates
+          render :json => all_rates.as_json, :status => :ok
+        elsif ups_rates
+          render :json => ups_rates.as_json, :status => :partial_content
+        elsif usps_rates
+          render :json => usps_rates.as_json, :status => :partial_content
+        else
+          render json: { error: :missing_destination_state, message: "You must provide a valid State for the shipping destination." }, status: :bad_request
+        end
+      # end
+    # rescue Timeout::Error
+    #   render :json => [], :status => :gateway_timeout
+    # end
```

Comment:
I know y'all are busy & code will _never_ be perfect, but please _please_ delete code when you've decided you no longer need it. Commenting it out and forgetting about it is a bad habit and will make it a lot harder on your team when they're reading your files a couple years later.